src/Common/Config/ConfigReloader.cpp

```diff
@@ -160,6 +163,13 @@ std::optional<ConfigProcessor::LoadedConfig> ConfigReloader::reloadIfNewer(bool
             return std::nullopt;
         }
         config_processor.savePreprocessedConfig(loaded_config, preprocessed_dir);
+#if USE_SSL
+        // load encryption codecs only for config.xml reloader
+        if (load_encryption_codecs == LoadEncryptionCodecs::Yes)
```

Comment:
Silly question but can we not unconditionally (always) load the encryption codecs? This would avoid the need for another member `load_encryption_codecs`. It sounds to me that loading encryption codecs is an idempotent operation.

Note that there is also a method `Configuration::tryLoad` which we could call. We could even add another parameter `bool log_error` to it which toggles if we trace or not.

Reply:
Repeated encryption codecs load will reset all previously loaded codecs, if no new codecs are found in config (because of aforementioned ```CompressionCodecEncrypted::Configuration::load``` behaviour). I'll change its logic, to assure idempotent behaviour, and remove this member.
```tryLoad``` does not help because no exception is raised in this case.